dartagnan/src/main/java/com/dat3m/dartagnan/solver/caat4wmm/ExecutionGraph.java

```diff
@@ -228,7 +227,10 @@ private RelationGraph createGraphFromRelation(Relation rel) {
 
         // ===== Filter special relations ======
         String name = rel.getNameOrTerm();
-        if (SPECIAL_RELS.contains(name)) {
+        if (cutRelations.contains(rel)) {
+            graph = new DynamicDefaultWMMGraph(name);
+        } else if (SPECIAL_RELS.contains(name)) {
+            // TODO: This case is unnecessary if we decide to cut the special relations upfront
```

Comment:
We are gonna need this for online solving anyway right? Also, IIRC the current solution is "theoretical unsound". I would say lets go for it